omod/src/main/resources/config.xml

```diff
@@ -60,6 +60,16 @@
 		<servlet-class>org.openmrs.module.fhir2.web.servlet.Fhir2SmartConfig</servlet-class>
 	</servlet>
 
+	<filter>
+		<filter-name>clientAuthenticationFilter</filter-name>
+		<filter-class>org.openmrs.module.fhir2.web.filter.ClientAuthenticationFilter</filter-class>
+	</filter>
+	<filter-mapping>
+		<filter-name>clientAuthenticationFilter</filter-name>
+		<url-pattern>/keycloak/*</url-pattern>
+		<url-pattern>/protected/*</url-pattern>
+	</filter-mapping>
```

Comment:
So what URLs do we actually want to protect?

Reply:
@ibacher I am not very sure about it that's why I left it unchanged

Reply:
The `url-pattern` determines which URLs this filter is run against. It should probably look very similar to the `AuthenticationFilter`, since they're ultimately protecting the same thing.

Reply:
Don't you think that it'll be different that the `AuthenticationFilter` because earlier we did not have any client (SMART App), we only had users

Reply:
No, because we're still protecting the same resources either way. That's what's important here.